src/System.Net.Primitives/src/System/Net/CookieContainer.cs

```diff
@@ -820,14 +820,16 @@ private void BuildCookieCollectionFromDomainMatches(Uri uri, bool isSecure, int
 
                 lock (pathList.SyncRoot)
                 {
-                    foreach (DictionaryEntry entry in pathList)
```

Comment:
I had to change this to no longer use `foreach` due to changing `PathList.GetEnumerator` from `public` to `internal`. Since I would have had to use the enumerator directly, I went ahead and changed it to the dictionary enumerator to avoid the `DictionaryEntry` box allocations.